### ISSUE TITLE
bitwarden_rs 1.9.1: build failure

### ISSUE THREAD

**Issue by amazari** [NixOS/nixpkgs#65839]
**Describe the bug**
Trying to install bitwarden_rs from master channel, lettre_email, a dependencies pulled by cargo, fails to compile:
```
error[E0283]: type annotations required: cannot resolve std::string::String: std::convert::AsRef<_>
   --> /build/bitwarden_rs-1.9.0-vendor/lettre_email/src/lib.rs:139:66
    |
139 |         self.header(("Content-Type", format!("{}", content_type).as_ref()))
    |                                                                  ^^^^^^

error: aborting due to previous error
```

**To Reproduce**
1. Add
```
services.bitwarden_rs = {
  enable = true;
};
```
to configuration.nix or
```
nix-env -iA bitwarden_rs
```

2. Rebuild configuration

**Expected behavior**
the `bitwarden_rs` binary is build and made available in PATH. Optionaly, it is ran as a systemd service.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.2.4, NixOS, 19.09pre186857.239fffc90d7 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre186857.239fffc90d7, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
 - system: `"x86_64-linux"`
 - host os: `Linux 5.2.4, NixOS, 19.09pre186857.239fffc90d7 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre186857.239fffc90d7, home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: services.bitwarden_rs
```
@marsam @msteen @elseym @c0bw3b @arcnmx

**Comment by arcnmx** [comment]
> /build/bitwarden_rs-1.9.0-vendor

The unstable channel has been stuck this past week, you'll need to use unstable-small or overrides or something to get 1.9.1 and [the patch](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/bitwarden_rs/cargo-lock-lettre.patch) I included with the 1.9.1 update (I doubt it will apply cleanly to 1.9.0 but it could be backported/recreated easily enough).

**Comment by amazari** [comment]
Thanks for your quick response and work on this package.

I'll use your solution in the interim before unstable is unstuck.

Guess this report can be closed.

Thanks again, have a good weekend.
